packages/@aws-cdk/aws-codepipeline-actions/lib/codecommit/source-action.ts

```diff
@@ -155,11 +180,23 @@ export class CodeCommitSourceAction extends Action {
       ],
     }));
 
+    if (this.props.codeBuildCloneOutput === true) {
```

Comment:
Can you actually fold this statement into the above one, considering their Resource is the same? Something like this:

```ts
      actions: [
        'codecommit:GetBranch',
        'codecommit:GetCommit',
        'codecommit:UploadArchive',
        'codecommit:GetUploadArchiveStatus',
        'codecommit:CancelUploadArchive',
        ...this.props.codeBuildCloneOutput ? ['codecommit:GetRepository'] : [],
      ],
```